charts/jenkins/templates/_helpers.tpl

```diff
@@ -207,6 +207,9 @@ Returns kubernetes pod template configuration as code
     value: {{ $value | quote }}
   {{- end }}
 {{- end }}
+  {{- if .Values.agent.id }}
+  id: {{ .Values.agent.id }}
```

Comment:
We might have an issue here with [additionalAgents](https://github.com/jenkinsci/helm-charts/tree/main/charts/jenkins#adding-pod-templates-using-additionalagents) as those inherit the settings.

So you could end up with different agent configurations using the same `agent.id`.

Could we relieve the user from specifying the id? e.g. by calculating a shasum of the agent name or it's whole config?
@timja I am not so much into the details of the kubernetes plugin do you have an idea of what could be used?

https://helm.sh/docs/howto/charts_tips_and_tricks/

Reply:
not sure

Reply:
Helm templates provide all functions from the sprig library.

http://masterminds.github.io/sprig/crypto.html

... so, a hash can be used without any hassles defined in the values.yaml file.

Should I close this and post a new PR?

Reply:
I'm testing locally by changing `_helpers.tpl`

```
id: {{ sha256sum (toYaml .Values.agent) }}
```

This results in the a rendered snippet like the following:

```
templates:
  - name: "default"
    id: 13c1f7ce7520a34cfdbd1dfdc297be8953a394bb11fb42822497a40cb21d49ab
```

This should provide a unique `agent.id` based on the entire contents of the `.Valeus.agent` key. No change to values.yaml necessary.

At this point, I'm not sure if I should open a new PR or just ammend this one with a new commit history.

Reply:
It's ok to update this one. Would make sense to also update the PR title.

Reply:
Could you check how this looks like for additionalAgents?

Reply:
Defining `additionalAgents` results in different ids for each one.

```
templates:
  - name: "default"
    id: 02c589989db49736dcf7e4b08f5b57a6dca297435bc20066f2623c1f78deb659
    ....
  - name: "maven"
    id: b4e6defdf6e92a8b5b9c8474acda539b1ac515d9ef54139919ddf641055618d9
    ....
  - name: "python"
    id: 5b90c727cc6187994a208667a472250b7997fee8bda82f81bc0a3da9a328b658
    ....
```

The generated ids are consistent with the contents of the agent Yaml from values.yaml. I will update the PR soon.

Thanks!

Reply:
does this cause any UI issues with the ID being so long? Is it displayed anywhere?

Reply:
As far as I can tell, the ID is not visible in the UI.